<a href="https://colab.research.google.com/github/welliver724/danl-m1-hw1-Welliver-Justin.ipynb/blob/main/danl-m1-hw2-Welliver-Justin.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
# Import pandas Library

import pandas as pd

In [5]:
# Import nhl DataFrame

nhl = pd.read_csv('https://bcdanl.github.io/data/NHL-1617.csv',
                  parse_dates = ["Born"])
nhl

<ipython-input-5-18ad355cc167>:3: UserWarning: Parsing dates in DD/MM/YYYY format when dayfirst=False (the default) was specified. This may lead to inconsistently parsed dates! Specify a format to ensure consistent parsing.
  nhl = pd.read_csv('https://bcdanl.github.io/data/NHL-1617.csv',


id_player       Born            City Cntry  Nat  Ht   Wt    Last_Name  \
0            1 1988-04-30        Hamilton   CAN  CAN  69  170       Abbott   
1            2 1987-02-25        Muskegon   USA  USA  74  218   Abdelkader   
2            3 1993-09-23       Stockholm   SWE  SWE  71  196        Aberg   
3            4 1991-01-12        Johnston   USA  USA  70  208      Acciari   
4            5 1992-04-30      Morristown   USA  USA  72  202     Agostino   
..         ...        ...             ...   ...  ...  ..  ...          ...   
883        884 1993-04-18        Huddinge   SWE  SWE  74  215    Zibanejad   
884        885 1987-01-09         Toronto   CAN  CAN  71  180  Zolnierczyk   
885        886 1987-01-09            Oslo   NOR  NOR  67  179   Zuccarello   
886        887 1992-01-16   Newport Beach   USA  USA  71  187       Zucker   
887        888 1995-05-15  St. Petersburg   RUS  RUS  73  224        Zykov   

    First_Name Position Team  GP   G   A    TOI  TOI_GP  
0      Spencer       LW  CHI   1   0   0    514    8.57  
1       Justin    LW/RW  DET  64   7  14  63969   16.65  
2       Pontus       LW  NSH  15   1   1  11102   12.33  
3         Noel        C  BOS  29   2   3  18047   10.23  
4        Kenny       LW  STL   7   1   2   5366   12.78  
..         ...      ...  ...  ..  ..  ..    ...     ...  
883       Mika     C/RW  NYR  56  14  23  57362   17.07  
884      Harry       LW  NSH  24   2   2  12776    8.87  
885       Mats  RW/C/LW  NYR  80  15  44  90378   18.83  
886      Jason    LW/RW  MIN  79  22  25  72455   15.28  
887   Valentin       LW  CAR   2   1   0    750    6.25  

[888 rows x 16 columns]

In [6]:
from google.colab import data_table
data_table.enable_dataframe_formatter()  # Enabling an interactive DataFrame display
nhl

id_player       Born            City Cntry  Nat  Ht   Wt    Last_Name  \
0            1 1988-04-30        Hamilton   CAN  CAN  69  170       Abbott   
1            2 1987-02-25        Muskegon   USA  USA  74  218   Abdelkader   
2            3 1993-09-23       Stockholm   SWE  SWE  71  196        Aberg   
3            4 1991-01-12        Johnston   USA  USA  70  208      Acciari   
4            5 1992-04-30      Morristown   USA  USA  72  202     Agostino   
..         ...        ...             ...   ...  ...  ..  ...          ...   
883        884 1993-04-18        Huddinge   SWE  SWE  74  215    Zibanejad   
884        885 1987-01-09         Toronto   CAN  CAN  71  180  Zolnierczyk   
885        886 1987-01-09            Oslo   NOR  NOR  67  179   Zuccarello   
886        887 1992-01-16   Newport Beach   USA  USA  71  187       Zucker   
887        888 1995-05-15  St. Petersburg   RUS  RUS  73  224        Zykov   

    First_Name Position Team  GP   G   A    TOI  TOI_GP  
0      Spencer       LW  CHI   1   0   0    514    8.57  
1       Justin    LW/RW  DET  64   7  14  63969   16.65  
2       Pontus       LW  NSH  15   1   1  11102   12.33  
3         Noel        C  BOS  29   2   3  18047   10.23  
4        Kenny       LW  STL   7   1   2   5366   12.78  
..         ...      ...  ...  ..  ..  ..    ...     ...  
883       Mika     C/RW  NYR  56  14  23  57362   17.07  
884      Harry       LW  NSH  24   2   2  12776    8.87  
885       Mats  RW/C/LW  NYR  80  15  44  90378   18.83  
886      Jason    LW/RW  MIN  79  22  25  72455   15.28  
887   Valentin       LW  CAR   2   1   0    750    6.25  

[888 rows x 16 columns]

In [9]:
# HW2 Q1 - How Many Players Had At Least Some Ice Time?

nhl[['TOI']].count()

TOI    888
dtype: int64

The total number of players who had some ice time is 888. I found this using .count() to count the number of non-missing values in a Series/DataFrame.

In [18]:
# HW2 Q2 - Who is the top scorer in terms of goals?

nhl.nlargest(1, 'G')

id_player       Born          City Cntry  Nat  Ht   Wt Last_Name  \
149        150 1987-07-08  Cole Harbour   CAN  CAN  71  200    Crosby   

    First_Name Position Team  GP   G   A    TOI  TOI_GP  
149     Sidney        C  PIT  75  44  45  89450   19.88

The top goal scorer in the DataFram is Cole Harbour. I'm sure there are many ways to get the top scorer in terms of goals, but I used the nlargest function. nlargest is good to get the first n observation ordered by a variable in descending order.

In [20]:
# HW2 Q3 - How can we count the number of NHL players for each country?

nhl[['Cntry']].value_counts()

Cntry
CAN      415
USA      242
SWE       77
RUS       39
CZE       31
FIN       30
CHE       12
SVK       10
DNK        9
DEU        6
AUT        5
FRA        4
NOR        2
ITA        1
LVA        1
HRV        1
SVN        1
EST        1
GBR        1
dtype: int64

.value_counts() counts the number of occurances of each unique value in a Series/DataFrame. I used .value_counts() to determine the number of NHL players that play for each country.

In [22]:
# HW2 Q4 - How many unique countries are in nhl.csv?

nhl[['Cntry']].nunique()

Cntry    19
dtype: int64

.nunique() counts the number of unique values in each variable in a Series/DataFrame. I used .nunique() to determine that there are 19 unique countries in the DataFrame.

In [27]:
# HW2 Q5 - Which three nationalities have the highest number of players?

df = nhl['Nat'].value_counts()
df.head(3)

CAN    413
USA    243
SWE     79
Name: Nat, dtype: int64

To find which 3 nationalities have the highest number of players, I first needed to count the number of occurances for each nationality in the Series/DataFrame. I did this by using .value_counts() on the 'Nat' column. I then used .head(3) to show only the top 3 largest values. I found that Canada, USA, and Sweden are the 3 top nationalities in the DataFrame.